pageserver/src/page_service.rs

```diff
@@ -915,6 +915,27 @@ where
             self.handle_basebackup_request(pgb, tenant_id, timeline_id, lsn, None, false, ctx)
                 .await?;
             pgb.write_message_noflush(&BeMessage::CommandComplete(b"SELECT 1"))?;
+        } else if query_string.starts_with("set_compute_spec_id ") {
+            let (_, params_raw) = query_string.split_at("set_compute_spec_id ".len());
```

Comment:
I guess this can be simplified as `if let Some((_, params_raw)) = query_string.split_once("set_compute_spec_id")`.